gradle/scripts/lib/common-dependencies.gradle

```diff
@@ -25,45 +25,96 @@ rootProject.ext {
+    if (key == 'boms') {
+        value.each {
+            def parsed = it.split(':')
+            managedDependencyVersions["${parsed[0]}:${parsed[1]}"] = parsed[2]
         }
-
-        imports {
-            if (rootProject.ext.dependenciesYaml.containsKey('boms')) {
-                dependenciesYaml['boms'].each {
-                    mavenBom "${it}"
+    } else {
+        def groupId = key
+        def artifact = value as Map
+        artifact.forEach { String artifactId, Map props ->
+            if (props.containsKey('version')) {
+                managedDependencyVersions["${groupId}:${artifactId}"] = "${props['version']}"
+            }
+            if (props.containsKey('exclusions')) {
+                props['exclusions'].each { String spec ->
+                    def parsed = spec.split(':')
+                    managedDependencyExclusions["${groupId}:${artifactId}"].add([
+                            group: parsed[0],
+                            module: parsed[1],
+                    ])
                 }
             }
         }
+    }
+}
+
+configure(projectsWithFlags('dependencies')) {
+    apply plugin: 'java-platform'
+
+    javaPlatform {
+        allowDependencies()
+    }
+
+    dependencies {
+        if (rootProject.ext.dependenciesYaml.containsKey('boms')) {
+            dependenciesYaml['boms'].each {
+                api platform("${it}")
+            }
+        }
 
-        dependencies {
+        constraints {
             rootProject.ext.dependenciesYaml.forEach { String key, value ->
                 if (key != 'boms') {
                     def groupId = key
                     def artifact = value as Map
                     artifact.forEach { String artifactId, Map props ->
                         if (props.containsKey('version')) {
-                            dependency("${groupId}:${artifactId}:${props['version']}") {
-                                if (props.containsKey('exclusions')) {
-                                    props['exclusions'].each { String spec ->
-                                        exclude spec
-                                    }
-                                }
-                            }
+                            api "${groupId}:${artifactId}:${props['version']}"
                         }
                     }
                 }
             }
         }
     }
+}
 
+allprojects {
     ext {
-        managedVersions = dependencyManagement.managedVersions
+        managedVersions = managedDependencyVersions
+    }
+}
+
+def dependencyProjects = projectsWithFlags('dependencies')
+if (!dependencyProjects.isEmpty()) {
+    def dependencyProject = dependencyProjects[0]
+    configure(projectsWithFlags('java')) {
+        // We need to use afterEvaluate because there is no way to guarantee configuration.canBeResolved and
+        // canBeConsumed are set otherwise.
+        afterEvaluate {
+            evaluationDependsOn(dependencyProject.path)
```

Comment:
Shouldn't we call `evaluationDependsOn(...)` from the outside of `afterEvaluate { .. }`?

Reply:
Doh yeah